pkg/restic/path_win.go

```diff
@@ -0,0 +1,37 @@
+//go:build windows
+// +build windows
+
+package restic
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+// toPathFilter converts a directory path to a path filter. Restic requires
+// these to be absolute, starting with a forward slash. On Windows, we convert
+// the drive letter to a single character and prepend a forward slash.
+func toPathFilter(path string) string {
+	cleanedPath := filepath.Clean(path)
+	// If clean results in an empty string, ignoring the volume, it returns "."
+	if cleanedPath != path+"." {
+		path = cleanedPath
+	}
+
+	before, after, found := strings.Cut(path, string(filepath.Separator))
+	if !found {
+		// e.g. if path is "C:"
+		before = path
+		after = ""
+	}
+
+	if len(before) == 2 && before[1] == ':' && before[0] >= 'A' && before[0] <= 'Z' {
```

Comment:
If I'm understanding correctly, the goal is to take a path that looks like `Z:/foo...` and convert it to something like `/Z/foo`?

I think we can simplify this a bit e.g.

```
func toRepoPath(path string) string {
   path := filepath.Clean(path) // not quite sure if we need this, we'll always call with absolute paths
   sepIdx := strings.Index(filepath.Separator)
   if sepIdx == -1 || sepIdx != 2 || path[1] != ':' {
      return path
   }
   return "/" + path[0] + "/" + path[3:]
}
```

also renamed to `toRepoPath` which maybe better represents what we're doing e.g. mapping a windows path format to a path as it's stored in the restic repo?